### ISSUE TITLE
[feature request] Backport smart fees from bitcoin core

### ISSUE THREAD

**Issue by runn1ng** [Bitcoin-ABC/bitcoin-abc#94]
Bitcoin core has a better fee algorithm in 0.15.0

See https://github.com/bitcoin/bitcoin/pull/10199 and other commits in the directory

It returns more accurate fees and returns fees even for small / big block targets. (With very empty chains, current algoritm returns -1 for small targets, and for targets >= 25)

The more accurate fees are maybe not important for emptier blockchains like Bitcoin Cash, *but* the API is already better, because it doesn't return -1 for either too small targets or too big targets.

**Comment by jli225** [comment]
NACK because:

1: It increases unnecessary risk without bringing in benefit.

2: I am afraid that the fee algorithms for 1mb and 8mb are not the same. Moreover, such algorithms are not useful at all for the fee market which only emerges after decades.

3. Bitcoin (cash) is not expected to have fee market in the future decades. Unnecessary fee algorithm will cause use confusion about the non-existence of fee market now.

**Comment by runn1ng** [comment]
In that case, it would make sense to remove regular, current fee estimates,
and make the RPC return a constant (for backwards compatibility).

It doesn't make sense to have the current fee estimation code, which is
inferior.

**Comment by jli225** [comment]
@runn1ng Things need time to be corrected. Have some patience.

**Comment by jli225** [comment]
@karel-3d Do you still want the dumb fee algorithm from BCore when you can simply pay the minimum fee to use Bitcoin Cash network? 
Time to stick with the true Bitcoin network, instead of the BCore chain which is backed by censorship and lies.

**Comment by deadalnix** [comment]
The long story short is that if nobody is going to do it, it is not going to get done.

**Comment by karel-3d** [comment]
@jli225 The current fee estimations are worse than the new "smart" ones from core. So it doesn't make sense to keep just the current ones.

Censorship on bitcoin core is out of the scope for this issue :)

**Comment by deadalnix** [comment]
@karel-3d I agree the fee estimation for ABC could do with some improvements. But I do not have the bandwidth to do it myself. If someone is willing to put in the work required to port the change and go through the review process, I'm happy to put in the work to do the review and to release this in ABC when it's ready.
